source/common/upstream/cluster_manager_impl.h

```diff
@@ -361,14 +367,38 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
   // This map is ordered so that config dumping is consistent.
   typedef std::map<std::string, ClusterDataPtr> ClusterMap;
 
+  struct PendingUpdates {
+    Event::TimerPtr timer_;
+    bool timer_enabled_{};
+    MonotonicTime last_updated_;
+    void enableTimer(const uint64_t timeout) {
```

Comment:
Can we add `ASSERT(!timer_enabled_);` and inverse below if possible (though seems not possible)?